### PR TITLE
[oneMKL][Stats][Spec] Simplify and clean up Summary Statistics domain

### PR DESCRIPTION
**Pull request by aelizaro**
* Simplify examples and make them pseudocode
* Remove unnecessary details about dataset

### CODE REVIEW

**Comment by aelizaro** [comment]
@paveldyakov, @andreyfe1, @iMartyan please, take a look

File: source/elements/oneMKL/source/domains/stats/onemkl_stats_usage_model.rst [hunk 1]
```diff
@@ -20,92 +20,32 @@ A typical algorithm for summary statistics is as follows:
...
         sycl::queue queue;
 
-        const size_t n_observations = 1000;
-        const size_t n_dims = 3;
+        constexpr std::size_t n_observations = 1000;
+        constexpr std::size_t n_dims = 3;
 
-        sycl::usm_allocator<float, sycl::usm::alloc::shared> allocator(queue);
+        // allocate USM memory for the dataset of the size n_observations * n_dims and fill it with any data
+        // allocate USM memory for the mean output of the size n_dims
```
Comment (andreyfe1) [inline source/elements/oneMKL/source/domains/stats/onemkl_stats_usage_model.rst:37]: USM is unified shared memory. So, USM memory stands for using "memory" twice :)

**Review by andreyfe1 — APPROVED** [review]
Looks good to me overall

**Comment by iMartyan** [comment]
lgtm

File: source/elements/oneMKL/source/domains/stats/onemkl_stats_usage_model.rst (hunk 1 above)
Comment (aelizaro) [reply source/elements/oneMKL/source/domains/stats/onemkl_stats_usage_model.rst]: good catch, fixed

**Review by andreyfe1 — APPROVED** [review]

**Pull request merged**